modules/lib/src/main/resources/assets/js/app/wizard/PageComponentsGridDragHandler.ts

```diff
@@ -173,6 +164,30 @@ export class PageComponentsGridDragHandler
         DragHelper.get().setDropAllowed(false);
     }
 
+    private isLayoutOverLayout(parentTreeItem: ComponentsTreeItem, draggedItem: ComponentsTreeItem): boolean {
+        const draggedItemType = draggedItem.getType();
+        const isFragment = draggedItemType instanceof FragmentComponentType;
+        const isLayout = !isFragment && draggedItemType instanceof LayoutComponentType;
+
+        if (isFragment || isLayout) {
+            if (this.isOverLayout(parentTreeItem) && (isLayout || this.isFragmentWithLayout(draggedItem))) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private isOverLayout(parentTreeItem: ComponentsTreeItem): boolean {
+        const grandParentPath: ComponentPath = this.helperFunctions.getPathByItem(parentTreeItem)?.getParentPath();
+        const grandParent: ComponentsTreeItem = grandParentPath ? this.helperFunctions.getItemByPath(grandParentPath) : null;
+        return grandParent?.getType() instanceof LayoutComponentType;
+    }
+
+    private isFragmentWithLayout(item: ComponentsTreeItem): boolean {
+        return item.getComponent()?.getDescription() === LayoutComponentType.get().getShortName();
+    }
```

Comment:
This method's logic is not limited to fragments, so it should have a different name (`isComponentWithLayout`?). If we absolutely want this to return boolean only for fragments, then it should have `instanceof FragmentComponentType` check inside.

Reply:
Updated